common/src/main/java/org/opensearch/ml/common/transport/search/MLSearchActionRequest.java

```diff
@@ -0,0 +1,101 @@
+package org.opensearch.ml.common.transport.search;
+
+import static org.opensearch.ml.common.CommonValue.VERSION_2_19_0;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+
+import org.opensearch.Version;
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.core.common.io.stream.InputStreamStreamInput;
+import org.opensearch.core.common.io.stream.OutputStreamStreamOutput;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+
+import lombok.Builder;
+import lombok.Getter;
+
+/**
+ * Represents an extended search action request that includes a tenant ID.
+ * This class allows OpenSearch to include a tenant ID in search requests,
+ * which is not natively supported in the standard {@link SearchRequest}.
+ */
+@Getter
+public class MLSearchActionRequest extends SearchRequest {
+    SearchRequest searchRequest;
+    String tenantId;
```

Comment:
A new MLSearchActionRequest class was added to wrap the original SearchRequest and include a tenantId field.So I want to understand does it impact other functionality using SearchRequest?

Reply:
Can you please give me an example which functionality are you referring to?

Reply:
I think the order matters, when did the SearchRequest converts to MLSearchActionRequest? I am worry about the tools and search pipelines that are using the search requests. if the transform is after them, it should be fine, but if that's before tools and search pipelines, does it impact them? 
for example
https://github.com/opensearch-project/ml-commons/blob/06a6021ffe5bc68d27db576895561831b7dc7aaa/ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/SearchIndexTool.java#L84

https://github.com/opensearch-project/ml-commons/blob/06a6021ffe5bc68d27db576895561831b7dc7aaa/search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/GenerativeQARequestProcessor.java#L51